### ISSUE TITLE
Exception on startup

### ISSUE THREAD

**Issue by tbenr** [Consensys-Incorporated/teku#7424]
```2023-08-16 13:04:01.204 INFO  - Loading initial state from https://bn.teku-nethermind-1.srv.dencun-devnet-8.ethpandaops.io/eth/v2/debug/beacon/states/finalized
2023-08-16 13:04:02.351 INFO  - Loaded initial state at epoch 20 (state root = 0xd1b98562e01b1bbf650ba86ac3423663d425d00ccc12ef588a268e83b2667137, block root = 0xd335cb550b4939478704c2bba453edb6ab82e00348a996a517a67eaeb0f9223c, block slot = 640).  Please ensure that the supplied initial state corresponds to the latest finalized block as of the start of epoch 20 (slot 640).
2023-08-16 13:04:02.581 ERROR - PLEASE FIX OR REPORT | Unexpected exception thrown for beaconchain-async-1
java.util.concurrent.CompletionException: java.util.NoSuchElementException: No value present
at java.util.concurrent.CompletableFuture.encodeRelay(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.completeRelay(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture$UniRelay.tryFire(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$propagateResult$3(SafeFuture.java:146) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.whenComplete(Unknown Source) ~[?:?]
at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:626) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at tech.pegasys.teku.infrastructure.async.SafeFuture.whenComplete(SafeFuture.java:33) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateResult(SafeFuture.java:143) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at tech.pegasys.teku.infrastructure.async.SafeFuture.propagateTo(SafeFuture.java:320) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:124) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.util.NoSuchElementException: No value present
at java.util.Optional.orElseThrow(Unknown Source) ~[?:?]
at tech.pegasys.teku.statetransition.forkchoice.ForkChoice.lambda$processHead$4(ForkChoice.java:223) ~[teku-ethereum-statetransition-develop.jar:23.8.0+7-gb54882aae2]
at tech.pegasys.teku.infrastructure.async.eventthread.AsyncRunnerEventThread.recordEventThreadIdAndExecute(AsyncRunnerEventThread.java:130) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at tech.pegasys.teku.infrastructure.async.eventthread.AsyncRunnerEventThread.lambda$doExecute$1(AsyncRunnerEventThread.java:113) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:82) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at tech.pegasys.teku.infrastructure.async.AsyncRunner.lambda$runAsync$2(AsyncRunner.java:47) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:74) ~[teku-infrastructure-async-develop.jar:23.8.0+7-gb54882aae2]
... 4 more
```

we need to check what can lead to have `final BeaconState justifiedState = justifiedCheckpointState.orElseThrow();` throwing in `processHead`, which means we were not able to retrieve any justified checkpoint state and that moment.

**Comment by rolfyone** [comment]
This could definitely happen when recentChainData hasn't set the store yet... I'll see if i can replicate that.

**Comment by tbenr** [comment]
> This could definitely happen when recentChainData hasn't set the store yet... I'll see if i can replicate that.

yes, I spent 5min exploring that circumstance and seemed not possible to me, at least in that code path. I might be wrong very easily.

**Comment by tbenr** [comment]
should be fixed by #7427
